test/Microsoft.Extensions.Internal.Test/StackTraceTest.cs

```diff
@@ -57,13 +75,69 @@ public void GetFrames_DoesNotFailForDynamicallyGeneratedAssemblies()
 
                 // Assert
                 Assert.Null(frames.First().FilePath);
-                Assert.Equal(nameof(GetFrames_DoesNotFailForDynamicallyGeneratedAssemblies), frames.Last().Method);
+                Assert.Equal(
+                    $"{typeof(StackTraceTest).GetTypeInfo().FullName}.{nameof(GetFrames_DoesNotFailForDynamicallyGeneratedAssemblies)}()",
+                    frames.Last().MethodDisplayInfo.ToString());
             }
         }
 
+        public static TheoryData GetMethodDisplayString_ReturnsTypeNameQualifiedMethodsData
```

Comment:
Done. It showed an issue with the code we have with Hosting today where the stack trace line we generate for nested types don't include the containing object.